### DISCUSSION TITLE
why isn't array validation working?

### DISCUSSION THREAD

**Discussion by prescriptionlifeline** [Laravel-Backpack/community-forum#1028] [Q&A (Help)] [answered]
Although I just got [FormRequest validation working](https://github.com/Laravel-Backpack/community-forum/discussions/1025) I was meanwhile experimenting with array validation and, for reasons unclear to me, it isn't working.

Here's my code:

```
        $rules = [
            'firstName' => ['required', 'string', 'max:128'],
            'lastName' => ['required', 'string', 'max:128'],
            'email' => ['required', 'string', 'lowercase', 'email', 'max:255'],
            'phoneCell' => ['nullable', 'string', 'max:25'],
            'phoneHome' => ['nullable', 'string', 'max:25'],
            'phoneWork' => ['nullable', 'string', 'max:25'],
            'address' => ['nullable', 'string', 'max:255'],
            'city' => ['nullable', 'string', 'max:255'],
            'state' => ['nullable', 'string', 'max:2'],
            'zip' => ['nullable', 'string', 'max:10'],
            'currentDues.*.amountPaid' => ['required'],
        ];

        CRUD::operation(['create'], function() {
            $rules['email'][] = new Email;
            $this->crud->setValidation($rules);
        });

        CRUD::operation(['update'], function() {
            $rules['email'][] = Rule::unique('users')->ignore(request()->input('id'));
            $this->crud->setValidation($rules);
        });
```
None of these rule are validating. Like I performed the update CRUD operation after deleting the first name field and I received no errors what-so-ever despite that field being required and not nullable.

Since FormRequest is working this isn't all that important but none-the-less it's now a point of curiosity for me.

**Comment by jcastroa87** [comment]
Hello @prescriptionlifeline as i understand you are handle this rules in Controller.

Can you share your controller code to check how is setup.

Thanks.

**Comment by jcastroa87** [comment]
Did you was abble to solve this? if you still need assistant please share the code to check the validation rules.

Cheers.

**Reply by prescriptionlifeline** [reply, marked as answer]
The issue was that I wasn't inheriting the `$rules` variable from the parent scope. So like instead of doing this:

```
CRUD::operation(['create'], function() {
```

I should have been doing this:

```
CRUD::operation(['create'], function() use ($rules) {
```

**Comment by jcastroa87** [comment]
Good to know, so i will close the issue, but please feel free to re-open or create a new one if needed.

Cheers.
